### PR TITLE
Mid level Refactor

### PR DESCRIPTION
**Pull request by ChineduOguejiofor**
#### What does this PR achieve?
- Updates Auth Routes to point to a new endpoint, 
- Updates Logic on user persistence
- Adds a spinner

### CODE REVIEW

File: src/components/signin/sagas.js [hunk 1]
```diff
@@ -27,16 +23,11 @@ const {
...
         const { user } = yield loginToFirebase.auth()
             .signInWithEmailAndPassword(payload.email, payload.password);
-        const getUserType = state => state.signIn.userType;
-        const userType = yield select(getUserType);
-        const userInfo = { userAuth: user, userType };
-        const config = { headers: { 'Content-Type': 'application/json' } };
-        const body = JSON.stringify(userInfo);
-        const response = yield axios.post(LOGIN_API_URL, body, config);
+        const userInfo = { userAuth: user };
+        const response = yield call([api, 'post'], authRequest.LOGIN, userInfo);
```
Comment (edbella) [inline src/components/signin/sagas.js]: You don't need "call" here. just yield (....) should suffice.

File: src/components/signin/sagas.js [hunk 2]
```diff
@@ -61,51 +52,23 @@ function* logoutUser() {
...
-        const LOGIN_API_URL = process.env.REACT_APP_USER_AUTH_API;
-        const getUserType = state => state.signIn.userType;
-        const userType = yield select(getUserType);
-        const apiResponse = channel();
-        yield loginToFirebase.auth().onAuthStateChanged(userData => {
-            prefetchExistingUser(userData, LOGIN_API_URL, apiResponse, userType);
-            return false;
+        const userAuth = yield getCurrentUser();
+        if (!userAuth) return;
+        const { data } = yield call([api, 'post'], authRequest.LOGIN, { userAuth });
```
Comment (edbella) [inline src/components/signin/sagas.js]: Same thing here as above. No need for "call".

File: src/components/signin/sagas.js [hunk 3]
```diff
@@ -61,51 +52,23 @@ function* logoutUser() {
...
+            payload: data.data.info,
+            type: VERIFY_SUCCESS,
         });
-        while (true) {
-            const action = yield take(apiResponse);
-            yield put(action);
-        }
     } catch (error) {
-        yield put(loginError(error));
+        yield put(logoutError(error));
```
Comment (edbella) [inline src/components/signin/sagas.js]: This should be loginError. The action being performed is to login an already signed-in user.

File: src/components/signup/sagas.js [hunk 4]
```diff
@@ -30,9 +30,7 @@ function* userSignUpEmail({ payload }) {
         const getUserType = state => state.signup.userType;
         const userType = yield select(getUserType);
         const userInfo = { fullName: name, userAuth: user, userType };
-        const config = { headers: { 'Content-Type': 'application/json' } };
-        const body = JSON.stringify(userInfo);
-        const request = yield axios.post(api, body, config);
+        const request = yield call([api, 'post'], authRequest.REGISTER, userInfo);
```
Comment (edbella) [inline src/components/signup/sagas.js]: Please remove all "call" from yield. Its not needed.

File: src/firebase.js [hunk 5]
```diff
@@ -52,11 +52,12 @@ export const uiConfig = {
     ],
 };
 
-if (process.env.NODE_ENV === 'test') {
-    firebase.auth().setPersistence(firebase.auth.Auth.Persistence.NONE);
-} else {
-    firebase.auth().setPersistence(firebase.auth.Auth.Persistence.LOCAL);
```
Comment (edbella) [inline src/firebase.js]: Why did you remove this?

File: src/App.jsx [hunk 6]
```diff
@@ -58,15 +46,18 @@ const routerSwitch = (
 
 export default function App() {
     runAos();
-    useFetching(verifyRequest);
-
+    useEffect(() => {
+        store.dispatch(verifyRequest());
+    });
```
Comment (edbella) [inline src/App.jsx]: Add an empty "[]" here so it runs once when the App component mounts. The way it is now will trigger it repeatedly.

File: src/components/signin/reducer.js [hunk 7]
```diff
@@ -51,15 +52,28 @@ export default (
...
+            loginError: false,
+            userObj: action.payload,
+        };
     case LOGIN_FAILURE:
         return {
             ...state,
             isAuthenticated: false,
+            isLoading: false,
             isLoggingIn: false,
+            loading: false,
```
Comment (edbella) [inline src/components/signin/reducer.js]: Why this? You already have "isLoading".

File: src/components/signin/sagas.js [hunk 8]
```diff
@@ -61,51 +52,23 @@ function* logoutUser() {
...
-        const apiResponse = channel();
-        yield loginToFirebase.auth().onAuthStateChanged(userData => {
-            prefetchExistingUser(userData, LOGIN_API_URL, apiResponse, userType);
-            return false;
+        const userAuth = yield getCurrentUser();
+        if (!userAuth) return;
+        const { data } = yield call([api, 'post'], authRequest.LOGIN, { userAuth });
+
+        yield put({
+            payload: data.data.info,
```
Comment (edbella) [inline src/components/signin/sagas.js]: Destructure "info" using this => "const { data: { info } } = response.data;"

File: src/components/signin/sagas.js (hunk 1 above)
Comment (ChineduOguejiofor) [reply src/components/signin/sagas.js]: I am making an API call here and passing some parameters,
using only yield distorts the flow and the returned item is not a response from the API call

File: src/components/signup/sagas.js (hunk 4 above)
Comment (ChineduOguejiofor) [reply src/components/signup/sagas.js]: same here
Thanks

File: src/firebase.js (hunk 5 above)
Comment (ChineduOguejiofor) [reply src/firebase.js]: Technically we don't need the LOCAL persistence anymore,
And I was also trying to test for PRODUCTION

File: src/App.jsx (hunk 6 above)
Comment (ChineduOguejiofor) [reply src/App.jsx]: updated 
Thanks

File: src/components/signin/sagas.js (hunk 8 above)
Comment (ChineduOguejiofor) [reply src/components/signin/sagas.js]: updated
Thanks

File: src/components/signin/sagas.js (hunk 3 above)
Comment (ChineduOguejiofor) [reply src/components/signin/sagas.js]: reverted
Thanks
